notifications_utils/testing/comparisons.py

```diff
@@ -0,0 +1,93 @@
+import re
+from functools import lru_cache
+from types import MappingProxyType
+from typing.re import Pattern
+
+
+class RestrictedAny:
+    """
+    Analogous to mock.ANY, this class takes an arbitrary callable in its constructor and the returned instance will
+    appear to "equal" anything that produces a truthy result when passed as an argument to the ``condition`` callable.
+
+    Useful when wanting to assert the contents of a larger structure but be more flexible for certain members, e.g.
+
+    # only care that second number is odd
+    >>> (4, 5, 6,) == (4, RestrictedAny(lambda x: x % 2), 6,)
+    True
+    >>> (4, 9, 6,) == (4, RestrictedAny(lambda x: x % 2), 6,)
+    True
+    """
+
+    def __init__(self, condition):
+        self._condition = condition
+
+    def __eq__(self, other):
+        return self._condition(other)
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}({self._condition})"
+
+    def __hash__(self):
+        return None
+
+
+class AnySupersetOf(RestrictedAny):
+    """
+    Instance will appear to "equal" any dictionary-like object that is a "superset" of the the constructor-supplied
+    ``subset_dict``, i.e. will ignore any keys present in the dictionary in question but missing from the reference
+    dict. e.g.
+
+    >>> [{"a": 123, "b": 456, "less": "predictabananas"}, 789] == [AnySupersetOf({"a": 123, "b": 456}), 789]
+    True
+    """
+
+    def __init__(self, subset_dict):
+        # take an immutable dict copy of supplied dict-like object
+        self._subset_dict = MappingProxyType(dict(subset_dict))
+        super().__init__(lambda other: self._subset_dict == {k: v for k, v in other.items() if k in self._subset_dict})
```

Comment:
What do you think about this to minimise use of lambdas?

```suggestion

    def _condition(self, other):
        return self._subset_dict == {k: v for k, v in other.items() if k in self._subset_dict}
```

Reply:
Yes that works doesn't it